### PR TITLE
Document attachments

### PR DESCRIPTION
**Pull request by blaisefonseca**

### CODE REVIEW

File: components/controls/edit/KmCkEditor.vue [hunk 1]
```diff
@@ -197,24 +197,30 @@ export default {
...
+      	if(value.url){
+      		self.onFileUpload({file:value})
+      	}
       }
+
+      this.$emit('onEditorReady', ed);
     },
     onEditorFocus( event, editor ) {
       // console.debug( 'Editor focused.', { event, editor } );
+        this.$emit('onEditorFocus', event, editor);
```
Comment (sbilodeau) [inline components/controls/edit/KmCkEditor.vue:209]: Your component has to declare all events it emits using `defineEmits()` in setup script or `emits:[]` in on classic component definition:

see: https://vuejs.org/guide/components/events.html#declaring-emitted-events

File: components/controls/edit/KmInputRichLstring.vue [hunk 2]
```diff
@@ -104,6 +104,9 @@ export default {
         const clean = useKmStorage().cleanDocument({...this.binding});
         this.$emit('update:modelValue', clean);
     },
+    onFileUpload({file}){
```
Comment (sbilodeau) [inline components/controls/edit/KmInputRichLstring.vue:107]: same, declare your events

**Review by sbilodeau — APPROVED** [review]
We were not use to it, but it's good practice to declare all events your component emits.

see: https://vuejs.org/guide/components/events.html#declaring-emitted-events

File: components/controls/edit/KmInputRichLstring.vue (hunk 2 above)
Comment (blaisefonseca) [reply components/controls/edit/KmInputRichLstring.vue:107]: yes , i the previous PR declared all the events, but probably u looked at the latter PR first #24

**Pull request merged**